### PR TITLE
Use compatProperties from config file

### PR DESCRIPTION
**Pull request by jw-msft**
- Remove compatPropertyNames
- Add compatProperties JSON_Value* to config info
- Send compatProperties json verbatim fr cfg in ainfo_req mqtt msg

### CODE REVIEW

File: src/extensions/agent_modules/cloud_communication/iothub_client/adu_core_interface/src/startup_msg_helper.c [hunk 1]
```diff
@@ -112,10 +112,7 @@ bool StartupMsg_AddCompatPropertyNames(JSON_Object* startupObj)
     bool success = false;
 
     JSON_Status jsonStatus = json_object_set_string(
-        startupObj,
-        ADUCITF_FIELDNAME_COMPAT_PROPERTY_NAMES,
-        IsNullOrEmpty(config->compatPropertyNames) ? DEFAULT_COMPAT_PROPERTY_NAMES_VALUE
-                                                   : config->compatPropertyNames);
```
Comment (Nox-MSFT) [inline src/extensions/agent_modules/cloud_communication/iothub_client/adu_core_interface/src/startup_msg_helper.c:118]: Do we still need this function?

File: src/utils/config_utils/tests/config_utils_ut.cpp [hunk 2]
```diff
@@ -30,7 +30,10 @@ static const char* validConfigContentStr =
         R"("aduShellTrustedUsers": ["adu","do"],)"
         R"("manufacturer": "device_info_manufacturer",)"
         R"("model": "device_info_model",)"
-        R"("compatPropertyNames": "manufacturer,model",)"
+        R"("compatProperties": { )"
+        R"(   "deviceManufacturer": "contoso", )"
+        R"(   "deviceModel": "camera", )"
```
Comment (Nox-MSFT) [inline src/utils/config_utils/tests/config_utils_ut.cpp:35]: Do we still need "manufacturer" and "model" property below?

**Review by Nox-MSFT — APPROVED** [review]

File: src/extensions/agent_modules/cloud_communication/iothub_client/adu_core_interface/src/startup_msg_helper.c (hunk 1 above)
Comment (jw-msft) [reply src/extensions/agent_modules/cloud_communication/iothub_client/adu_core_interface/src/startup_msg_helper.c:118]: no but will remove along when remove cloud_communication/iothub_client

File: src/utils/config_utils/tests/config_utils_ut.cpp (hunk 2 above)
Comment (jw-msft) [reply src/utils/config_utils/tests/config_utils_ut.cpp:35]: no, not needed. removing

**Pull request merged**